### PR TITLE
Download progress

### PR DESCRIPTION
**Pull request by taubenangriff**
Adds progress to async downloads

### CODE REVIEW

File: ModManager_Classes/GithubIntegration/GithubDownloader.cs [hunk 1]
```diff
@@ -8,12 +8,22 @@
...
+        public String DownloadDestination = String.Empty;
+    }
+
     public class GithubDownloader
     {
         private readonly String DOWNLOAD_DIRECTORY;
 
         private static GitHubClient GithubClient = new GitHubClient(new ProductHeaderValue("iModYourAnno"));
 
+        public static int DownloadBufferSize { get; } = 81920;
```
Comment (EtienneDx) [inline ModManager_Classes/GithubIntegration/GithubDownloader.cs:25]: Does this restrict the dl to 80 Mb?

**Review by EtienneDx — COMMENTED** [review]
Appart from the 80 Mb (which is probably fine but idk) looks great

File: ModManager_Classes/Utils/ModLoaderInstaller.cs [hunk 2]
```diff
@@ -39,11 +39,12 @@ public async Task InstallAsync()
             }
 
             var modloaderRepo = new GithubRepoInfo() { Name = "anno1800-mod-loader", Owner = "xforce" };
-            string downloadResult = await GithubDownloader!.DownloadReleaseAsync(modloaderRepo, "loader.zip");
+            var downloadResult = await GithubDownloader!.DownloadReleaseAsync(modloaderRepo, "loader.zip", progress);
```
Comment (jakobharder) [inline ModManager_Classes/Utils/ModLoaderInstaller.cs:42]: GithubDownload is created in the constructor. It's better to make it non-nullable than using `!`.

File: ModManager_Classes/GithubIntegration/GithubDownloader.cs (hunk 1 above)
Comment (taubenangriff) [reply ModManager_Classes/GithubIntegration/GithubDownloader.cs:25]: No, it's the buffer size for the download streamcopy.

The copy method works like this: 

while DownloadStream has bytes left
read buffersize amount of bytes from the source stream (http response)
copy buffer to target stream (filestream)

File: ModManager_Classes/Utils/ModLoaderInstaller.cs (hunk 2 above)
Comment (taubenangriff) [reply ModManager_Classes/Utils/ModLoaderInstaller.cs]: nice catch, resolved in [c77d9a0](https://github.com/anno-mods/iModYourAnno/pull/69/commits/773d9a07832adab927f33fb0382f5fd505624a7e)

**Review by jakobharder — APPROVED** [review]

File: ModManager_Classes/GithubIntegration/GithubDownloader.cs (hunk 1 above)
Comment (EtienneDx) [reply ModManager_Classes/GithubIntegration/GithubDownloader.cs:25]: all good then

**Pull request merged**